common/cuda_hip/matrix/batch_ell_kernels.hpp.inc

```diff
@@ -125,3 +125,43 @@ __global__ __launch_bounds__(
                        x_b.values);
     }
 }
+
+
+template <typename ValueType, typename IndexType>
+__device__ __forceinline__ void scale(
+    const ValueType* const __restrict__ col_scale,
+    const ValueType* const __restrict__ row_scale,
+    const gko::batch::matrix::ell::batch_item<ValueType, IndexType>& mat)
+{
+    for (int tidx = threadIdx.x; tidx < mat.num_rows; tidx += blockDim.x) {
```

Comment:
Csr uses a warp-per-row parallelization, why does this use thread-per-row again?

Reply:
Csr matrices tend to be more unbalanced, while Ell matrices are balanced.